src/Model/ContextKeys.re

```diff
@@ -77,7 +77,19 @@ let other =
         | {completions, _} when Completions.isActive(completions) => true
         | _ => false,
       ),
-      bool("sneakMode", state => Sneak.isActive(state.sneak)),
+      bool("isLinux", _state =>
+        Revery.Environment.os == Revery.Environment.Linux
+      ),
+      bool("isMax", _state => Revery.Environment.os == Revery.Environment.Mac),
+      bool("isWin", _state =>
+        Revery.Environment.os == Revery.Environment.Windows
+      ),
+      bool("oni.sneakMode", state => Sneak.isActive(state.sneak)),
+      bool("oni.zenMode", state => state.zenMode),
+      bool("oni.keyDisplayerEnabled", state => state.keyDisplayer != None),
+      bool("oni.symLinkExists", _state =>
+        Sys.file_exists("/usr/local/bin/oni2")
+      ),
```

Comment:
An example of an impure context key, which is still possible, though not ideal.

Reply:
I'm concerned about this - it means we're doing blocking file I/O on potentially every key press or render. Is there another way we could model this, such that it only gets called 'lazily' - ie, when something is using `oni.symLinkExists`? I believe this is only needed when we open the quick menu.

Alternatively - we could store `symLinkExists` in the state somewhere, and check it. It's not perfect, because it could be out of sync, but I think the performance trade-off is worth it.

The concern I have is that synchronous file I/O on our critical rendering / input paths can lead to unpredictable performance characteristics, especially on Windows.

Reply:
(I believe this was less of an issue before, since it only was ran when the command palette was open)

Reply:
Ah, yeah, good catch. Since context keys can be extended, we could also just add it on-demand when retrieving the command palette. But ideally we'd not have the commands using this defined globally either. The command palette retrieval also changes a bit with the menu contribituons PR, but I'll look into a solution that will work for both.

Reply:
Fixed in https://github.com/onivim/oni2/pull/1595/commits/4590021853321fe5915b2490f882df02fffb8897